src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,189 @@
+import sas_testcase
+import hashlib
+from datetime import datetime, timedelta
+
+class FullActivityDumpMessageTestcase(sas_testcase.SasTestCase):
+
+    def setUp(self):
+        self._sas, self._sas_admin = sas.GetTestingSas()
+        self._sas_admin.Reset()
+
+    def tearDown(self):
+        pass
+    
+    def assertCbsdRecord(self, registration_request, grant_request, grant_response, recordData):
+        for index, device in enumerate(registration_request):
+            record_id = 'cbsd/'+ device['fccId']+'/'+ hashlib.sha1(device['cbsdSerialNumber']).hexdigest()
+            cbsd_record = [record['registration'] for record in recordData if record['id'] == record_id]
+            self.assertEqual(1, len(cbsd_record))
+            self.assertEqual(device['cbsdCategory'], cbsd_record[0]['cbsdCategory'])
+            self.assertEqual(device['fccId'], cbsd_record[0]['fccId'])
+            self.assertEqual(device['airInterface']['radioTechnology'],\
+                             cbsd_record[0]['airInterface']['radioTechnology'])
+            self.assertEqual(device['installationParam']['latitude'],\
+                             cbsd_record[0]['installationParam']['latitude'])
+            self.assertEqual(device['installationParam']['longitude'],\
+                             cbsd_record[0]['installationParam']['longitude'])
+            self.assertEqual(device['installationParam']['height'],\
+                             cbsd_record[0]['installationParam']['height'])
+            self.assertEqual(device['installationParam']['heightType'], \
+                             cbsd_record[0]['installationParam']['heightType'])
+            self.assertEqual(device['installationParam']['indoorDeployment'], \
+                             cbsd_record[0]['installationParam']['indoorDeployment'])
+            self.assertEqual(device['installationParam']['antennaGain'], \
+                             cbsd_record[0]['installationParam']['antennaGain'])     
+            self.assertEqual(device['installationParam']['antennaAzimuth'], \
+                             cbsd_record[0]['installationParam']['antennaAzimuth'])        
+            self.assertEqual(device['installationParam']['antennaDowntilt'], \
+                             cbsd_record[0]['installationParam']['antennaDowntilt'])       
+            self.assertEqual(device['installationParam']['antennaBeamwidth'], \
+                             cbsd_record[0]['installationParam']['antennaBeamwidth'])    
+                         
+            # Get grants by cbsd_id
+            grants_of_cbsd = [cbsd['grants'] for cbsd in recordData if cbsd['id'] == record_id]
+            self.assertEqual(1, len(grants_of_cbsd))
+            self.assertTrue('id' in grants_of_cbsd[0])
+            # Verify the Grant Of the Cbsd
+            if grants_of_cbsd[0]['requestedOperationParam'] is not None :
+                self.assertEqual(grants_of_cbsd[0]['requestedOperationParam']['maxEirp'],\
```

Comment:
Despite the name, these actually do not need to be equal :) I believe the SAS is allowed to put any value, up to the registered max EIRP (careful with the units) for the CBSD and as low as -137 dBm/MHz (defined by the spec). (Some) more details can be found in the SAS admins agreement.

Similarly the frequency range should be a valid but need not have any particular value. "Valid" in this case means that it's in 3550-3700 and aligns with multiples of 5 MHz.